jupyter_core/utils/__init__.py

```diff
@@ -182,6 +177,17 @@ def ensure_event_loop(prefer_selector_loop: bool = False) -> asyncio.AbstractEve
         loop = asyncio.get_running_loop()
     except RuntimeError:
         if sys.platform == "win32" and prefer_selector_loop:
+            if (3, 14) <= sys.version_info < (3, 15):
+                # ignore deprecation only for 3.14 and revisit later.
+                with warnings.catch_warnings():
+                    warnings.simplefilter(
```

Comment:
I think you want

```suggestion
                    warnings.filterwarnings(
```

for the `message` argument

and maybe just filter on `.*WindowsSelectorEventLoopPolicy.*` to avoid missing irrelevant changes in language, or if removal happens to be kicked down the road to 3.17, etc.